remapping_commands.sh

```diff
@@ -0,0 +1,189 @@
+#!/bin/bash
+#requires vcf2bed, bowtie2, samtools, bedtools, reverse_strand.py (uses bamnostic) and write_header.py
+
+#This basic pipeline allows you to input two assemblies and a variant vcf file, 
+#and remap the variants to the new genome
+
+
+#example command:
+#time ./remapping_commands.sh -g droso_dm3.fasta -n GCA_000001215.4_Release_6_plus_ISO1_MT_genomic.fna -a GCA_000001215.4 -v droso_variants_renamed.vcf -o test.vcf
+
+
+#To be added:
+# -support for indels
+# -find a way to extract new assembly accession so it's not required as input?
+
+#Completed:
+# -filter only SNPs (not indels) using bcftools filter
+# -copy filter etc columns from original vcf
+# -copy vcf header to the new vcf file, but regenerate contigs from the list of contigs in new assembly, and change reference tag
+# -take into account reverse strand
+# -create a command line version where you input file names
+# -% of remapped variants calculation
+
+
+oldgenome=''
+newgenome=''
+newgenomeaccession=''
+vcffile=''
+outfile=''
+
+while getopts 'g:n:a:v:o:' flag; do
+	case "${flag}" in
+		g) oldgenome="${OPTARG}";;
+		n) newgenome="${OPTARG}" ;;
+		a) newgenomeaccession="${OPTARG}" ;;
+		v) vcffile="${OPTARG}" ;;
+		o) outfile="${OPTARG}" ;;
+	esac
+done
+
+
+#1) Flanking sequence generation
+echo "-----------------------1) Flanking sequence generation-----------------------"
+#generate chrom sizes:
+samtools faidx $oldgenome 
+cut -f1,2 "$oldgenome".fai > "$oldgenome".chrom.sizes
+
+#filter SNPs only
+bcftools filter -i 'TYPE="snp"' $vcffile > snps_only.vcf
+
+#store header
+awk '$1 ~ /^#/ {print $0}' snps_only.vcf > vcf_header.txt
+
+#convert vcf to bed:
+vcf2bed < snps_only.vcf > variants.bed
+#the actual position of the variant is the second coordinate
+#rm snps_only.vcf
+
+#generate the flanking sequence intervals
+bedtools flank -i variants.bed -g "$oldgenome".chrom.sizes -b 50 > flanking.bed
+rm variants.bed "$oldgenome".chrom.sizes
+
+#fix rows: put both flanking sequences on 1 row = 1 read and add 1 to start position so the read is 100 bases:
```

Comment:
Could you please also document briefly what it looked like before, to understand better the reason for these transformations?

Reply:
Of course, I have now added an explanation directly into the remapping_commands.sh file.